### PR TITLE
support pending channels

### PR DESCRIPTION
**Pull request by jessopb**

### CODE REVIEW

File: src/redux/reducers/claims.js [hunk 1]
```diff
@@ -112,18 +114,19 @@ function handleClaimAction(state: State, action: any): State {
...
     const { claimsInChannel, stream, channel } = resolveResponse;
-    if (claimsInChannel) {
-      channelClaimCounts[url] = claimsInChannel;
-      channelClaimCounts[channel.canonical_url] = claimsInChannel;
-    }
 
     if (stream) {
-      byId[stream.claim_id] = stream;
+      if (pendingIds.includes(stream.claim_id)) {
+        byId[stream.claim_id] = mergeClaim(stream, byId[stream.claim_id]);// merge them
```
Comment (jessopb) [inline src/redux/reducers/claims.js:126]: comment

File: src/redux/reducers/claims.js [hunk 2]
```diff
@@ -198,47 +207,37 @@ reducers[ACTIONS.FETCH_CLAIM_LIST_MINE_COMPLETED] = (state: State, action: any):
...
+        pendingIdSet.add(claimId);
+      } else if (!resolve && pendingIdSet.has(claimId) && claim.confirmations > 0) {
+        pendingIdSet.delete(claimId);
+      }
+      if (pendingIds.includes(claimId)) {
+        byId[claimId] = mergeClaim(claim, byId[claimId]);
       } else {
-        byId[claimId] = claim;
-        byUri[uri] = claimId;
+        byId[claimId] = claim;// just add
```
Comment (jessopb) [inline src/redux/reducers/claims.js:228]: comment

File: src/redux/reducers/claims.js [hunk 3]
```diff
@@ -275,18 +277,11 @@ reducers[ACTIONS.FETCH_CHANNEL_LIST_COMPLETED] = (state: State, action: any): St
 
       // $FlowFixMe
       myChannelClaims.add(claimId);
-      if (!byId[claimId]) {
+      if (!pendingIds.some(c => c === claimId)) {
         byId[claimId] = claim;
       }
-
+      myChannelUrls.push(permanentUrl);
```
Comment (jessopb) [inline src/redux/reducers/claims.js:283]: check flow error (can't .push null, but set to array above)

File: src/redux/reducers/claims.js (hunk 2 above)
Comment (seanyesmunt) [reply src/redux/reducers/claims.js:228]: comment

**Comment by jessopb** [comment]
> I think `myChannelUrls` could be removed.
> 
> `selectChannelUrls` could just look like
> 
> ```
> export const selectMyChannelUrls = createSelector(
>   selectState,
>   state => state.myChannelClaims.map(claim => claim.canonical_url)
> );
> ```

Changed this (myChannelClaims is actually [claimId, ...]). fall back on permanent_url for new channels still pending.

**Pull request merged**